images/singleuser-sample/Dockerfile

```diff
@@ -1,26 +1,61 @@
-FROM jupyter/base-notebook:latest
-# Built from... https://hub.docker.com/r/jupyter/base-notebook/
-#               https://github.com/jupyter/docker-stacks/blob/HEAD/base-notebook/Dockerfile
-# Built from... Ubuntu 20.04
+# The build stage
+# ---------------
+FROM python:3.9-bullseye as build-stage
 
-# VULN_SCAN_TIME=2022-08-08_07:54:05
+# VULN_SCAN_TIME=
+
+WORKDIR /build-stage
+
+# Build wheels for packages that require gcc or other build dependencies and
+# lack wheels either for amd64 or aarch64.
+COPY requirements.txt requirements.txt
+RUN pip wheel -r requirements.txt
```

Comment:
It is better to avoid building wheels here from a final size perspective, because when you copy and then install in another stage, the copy + install layers will be separate.

I think we won't need a build stage for this image actually.